takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/jdt/FilerImpl.java

```diff
@@ -44,6 +46,7 @@
   private final boolean incremental;
 
   private final Set<URI> createdResources = new HashSet<>();
+  private final Set<File> writtenFiles = new HashSet<>();
```

Comment:
Since you're retaining state across the run of the filer, you should probably document how it's used.  One filer instance per round?  Or one across all rounds.